codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java

```diff
@@ -31,9 +34,12 @@
 
 public final class AuthSchemeSpecUtils {
     private final IntermediateModel intermediateModel;
+    private final Set<String> allowedEndpointAuthSchemeParams;
 
     public AuthSchemeSpecUtils(IntermediateModel intermediateModel) {
         this.intermediateModel = intermediateModel;
+        this.allowedEndpointAuthSchemeParams = Collections.unmodifiableSet(
```

Comment:
why both a new HashSet and unmodifiableSet

Reply:
Because `getAllowedEndpointAuthSchemeParams` returns a `List<String>`

Reply:
And `unmodifiableSet` to make the `Set` immutable.